app.rb

```diff
@@ -0,0 +1,300 @@
+    # sometime returns an error: "Error locking accountRecordId='***'"
+    subscription.create(user, reason, comment, nil, true, options)
+  rescue Exception => e
+    puts e.message
+  end
+end
+
+#
+# Sinatra handlers
+#
+
+get '/' do
+  erb :index
+end
+
+post '/charge' do
+  # Create an account
+  account = create_kb_account(user, reason, comment, options)
+
+  # Add a payment method associated
+  create_kb_payment_method(account, params['paypageRegistrationId'], user, reason, comment, options)
+
+  # Add a subscription
+  create_subscription(account, user, reason, comment, options)
+
+  # Retrieve the invoice
+  @invoice = account.invoices(true, options).first
+  @accountId = account.account_id
+
+  erb :charge
+end
+
+__END__
+
+@@ layout
+  <!DOCTYPE html>
+  <html>
+  <head>
+      <meta charset="UTF-8">
+      <title>Litle Demo page</title>
+      <style>
+          body {
+              font-size:10pt;
+          }
+          .checkout {
+              background-color:lightgrey;
+              width: 50%;
+          }
+          #payframe {
+              background-color:darkseagreen;
+          }
+          .testFieldTable {
+              background-color:lightgrey;
+          }
+          #submitId {
+              font-weight:bold;
+              font-size:12pt;
+          }
+          form#fCheckout {
+          }
+      </style>
+      <script src="https://code.jquery.com/jquery-3.1.1.min.js" type="text/javascript"></script>
+      <script src="https://request-prelive.np-securepaypage-litle.com/LitlePayPage/js/payframe-client.min.js" type="text/javascript"></script>
+  </head>
+  <body>
+    <%= yield %>
+  </body>
+  </html>
+
+@@index
+  <span class="image"><img src="https://drive.google.com/uc?&amp;id=0Bw8rymjWckBHT3dKd0U3a1RfcUE&amp;w=960&amp;h=480" alt="uc?&amp;id=0Bw8rymjWckBHT3dKd0U3a1RfcUE&amp;w=960&amp;h=480"></span>
+  <article>
+      <label class="amount">
+        <span>Sports car, 30 days trial for only $10.00!</span>
+      </label>
+  </article>
+  <div class="checkout">
+        <h2>Checkout Form</h2>
+        <form method=post id="fCheckout" name="fCheckout" action="/charge">
+            <table>
+                <tr><td colspan="2">
+                    <div id="payframe">
+                    </div>
+                </td></tr>
+                <tr><td>Paypage Registration ID</td><td><input type="text" id="paypageRegistrationId" name="paypageRegistrationId" readOnly="true"/> <--Hidden</td></tr>
+                <tr><td>Bin</td><td><input type="text" id="bin" name="bin" readOnly="true"/> <--Hidden</td></tr>
+                <tr><td></td><td align="right"><input type="submit" id="submitId">Check out</button></td></tr>
+            </table>
+        </form>
+        <br/>
+        <h3>Test Input Fields</h3>
+        <table class="testFieldTable">
+            <tr>
+                <td>Paypage ID</td><td><input type="text" id="request$paypageId" name="request$paypageId" value="a2y4o6m8k0" disabled/></td>
```

Comment:
This should be configurable as it depends on the merchant - maybe introduce an environment variable like `KB_URL` (and update the README)?